docs/sanic/config.rst

```diff
@@ -47,8 +47,85 @@ Then the above variable would be `MYAPP_REQUEST_TIMEOUT`. If you want to disable
 
     app = Sanic(__name__, load_env=False)
 
-From an Object
-~~~~~~~~~~~~~~
+From file, dict, any object (having __dict__ attribute).
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You may wish to keep Your app configurations somewhere in a python file, or in some dictoionary, or even parhaps in some other object.
+In order to load configuration from those You can use app.upload_config() method.
+
+
+- From file
+So lets say You have my_config.py file like:
+
+.. code-block:: python
+
+    # my_config.py
+    A = 1
+    B = 2
+
+Loading this config from this file is as easy as:
+
+.. code-block:: python
+
+    app.update_config("/path/to/my_config.py")
+ 
+You can also use environment variables here.
+
+Lets say you have environment variable like:
+
+.. code-block:: shell
+
+    > my_path="/path/to"
+    
+So then to make use of it just:  
+
+.. code-block:: python
+
+    app.update_config("${my_path}/my_config.py")
+
+Just remember you have to provide environment variables in format ${environment_variable}
+and that $environment_variable is not expanded (is treated as "plain" text).
+ 
+- From dict
+
+You can upload app config by providing dict holding settings:
+
+.. code-block:: python
+
+    d = {"A": 1, "B": 2}
+    
+    app.update_config(d)
+
+- From an object (having __dict__ attribute)
+
+You can upload app config by providing any object holding settings,
+but in such case __dict__ attribute of this object will be used as dict holding settings.
+
+For example:
+
+.. code-block:: python
+
+    class C:
+        A = 1
+        B = 2
+        
+    app.update_config(C)
+
+or
+
+.. code-block:: python
+
+    c = C()
+    
+    app.update_config(C)
+    
+etc.
+
+(If You are not shure just see e.g. what C.__dict__ contains)
+
+
+From an Object (Deprecated, will be removed in version 21.3)
```

Comment:
Is there nether method to point out this as deprecated ?

Reply:
Here are my edits to this:

```
From file, dict, or any object (having __dict__ attribute).
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

You can store app configurations in: (1) a Python file, (2) a dictionary, or (3) in some other type of custom object.

In order to load configuration from ove of those, you can use ``app.upload_config()``.

From file
---------

Let's say you have ``my_config.py`` file that looks like this:

.. code-block:: python

    # my_config.py
    A = 1
    B = 2

Loading config from this file is as easy as:

.. code-block:: python

    app.update_config("/path/to/my_config.py")
 
You can also use environment variables in the path name here.

Let's say you have an environment variable like this:

.. code-block:: shell

    $ export my_path="/path/to"
    
Then you can use it like this:

.. code-block:: python

    app.update_config("${my_path}/my_config.py")

.. note::

    Just remember that you have to provide environment variables in the format ${environment_variable} and that $environment_variable is not expanded (is treated as "plain" text).

From dict
---------

You can also set your app config by providing a ``dict``:

.. code-block:: python

    d = {"A": 1, "B": 2}
    
    app.update_config(d)

From _any_ object
=================

App config can be taken from an object. Internally, it uses ``__dict__`` to retrieve keys and values.

For example, pass the class:

.. code-block:: python

    class C:
        A = 1
        B = 2
        
    app.update_config(C)

or, it can be instantiated:

.. code-block:: python

    c = C()
    
    app.update_config(c)


From an Object 
==============

.. note::
    
    (Deprecated, and will be removed in version 21.3)
```

Reply:
I would suggest carrying through the note about deprecation.